inc/updates.php

```diff
@@ -198,6 +194,146 @@ function bh_sync_plugin_major_auto_core_update_option( $old_value, $value ) {
+		return;
+	}
+
+	bh_sync_plugin_update_settings();
+}
+
+add_action( 'deleted_plugin', 'bh_deleted_plugin', 10, 2 );
+
+/**
+ * Updates the WordPress Core theme auto-update option when a theme is deleted.
+ *
+ * @param string $stylesheet Stylesheet of the theme to delete.
+ * @param bool   $deleted    Whether the theme deletion was successful.
+ */
+function bh_deleted_theme( $stylesheet, $deleted ) {
+	if ( ! $deleted ) {
+		return;
+	}
+
+	// When the option is disabled in the plugin, theme auto-updates are manually managed in the WordPress Core UI.
+	if ( 'false' === get_site_option( 'auto_update_theme', 'true' ) ) {
+		return;
+	}
+
+	bh_sync_theme_update_settings();
+}
+
+add_action( 'deleted_theme', 'bh_deleted_plugin', 10, 2 );
+
+/**
+ * Updates the WordPress Core theme auto-update option when a theme is deleted.
+ *
+ * This is a backwards compatibility version of `bh_deleted_theme`, which uses the `deleted_theme` action hook that was
+ * added in WordPress 5.8.0.
+ */
+function bh_delete_site_transient_update_themes() {
+	global $wp_version;
+
+	// The `deleted_theme` hook was introduced in WordPress 5.8.
+	if ( version_compare( $wp_version, '5.8.0', '>=' ) ) {
+		return;
+	}
+
+	// When the option is disabled in the plugin, theme auto-updates are manually managed in the WordPress Core UI.
+	if ( 'false' === get_site_option( 'auto_update_theme', 'true' ) ) {
+		return;
+	}
+
+	bh_sync_theme_update_settings();
+}
+
+add_action( 'delete_site_transient_update_themes', 'bh_delete_site_transient_update_themes', 10, 0 );
+
+/**
+ * Updates the WordPress Core options for plugin and theme auto-updates when one is updated.
+ *
+ * @param WP_Upgrader $upgrader WP_Upgrader instance. In other contexts, $this, might be a
+ *                              Theme_Upgrader, Plugin_Upgrader, Core_Upgrade, or Language_Pack_Upgrader instance.
+ * @param array       $hook_extra {
+ *     Array of bulk item update data.
+ *
+ *     @type string $action       Type of action. Default 'update'.
+ *     @type string $type         Type of update process. Accepts 'plugin', 'theme', 'translation', or 'core'.
+ *     @type bool   $bulk         Whether the update process is a bulk update. Default true.
+ *     @type array  $plugins      Array of the basename paths of the plugins' main files.
+ *     @type array  $themes       The theme slugs.
+ *     @type array  $translations {
+ *         Array of translations update data.
+ *
+ *         @type string $language The locale the translation is for.
+ *         @type string $type     Type of translation. Accepts 'plugin', 'theme', or 'core'.
+ *         @type string $slug     Text domain the translation is for. The slug of a theme/plugin or
+ *                                'default' for core translations.
+ *         @type string $version  The version of a theme, plugin, or core.
+ *     }
+ * }
+ */
+function bh_upgrader_process_complete( $upgrader, $hook_extra ) {
```

Comment:
This function will update the option for any install or update action on a plugin or theme.

I'm not entirely convinced that we need to verify the option with every plugin or theme update. But, if we want to be sure the lists stay up to date then we should check on updates.

The main scenario that would benefit from this is when someone uploads a plugin or theme manually. No action hooks are executed, so the plugin or theme would never make its way into the update list. And if the site owner deactivates/removes the Bluehost plugin without adding or deleting any plugins/themes, the auto-update list will be stale and not all inclusive.